SQ8H search failed run on base dataset sift-1b

**Describe the bug**
```
10:41:58  2020-03-18:02:41:57,993 INFO     [runner.py:196] Start query, query params: top-k: 1, nq: 200, actually length of vectors: 200
10:41:58  2020-03-18:02:41:57,993 INFO     [runner.py:198] Start run query, run 1 of 2
10:42:11  2020-03-18:02:42:10,608 ERROR    [client.py:71] out of memory: CPU2GPU:1
10:42:11  2020-03-18:02:42:10,608 ERROR    [main.py:69] Status not ok
10:42:11  2020-03-18:02:42:10,611 ERROR    [main.py:70] Traceback (most recent call last):
10:42:11    File "main.py", line 67, in queue_worker
10:42:11      runner.run(run_type, collection)
10:42:11    File "/home/jenkins/agent/workspace/milvus-debug/milvus_benchmark/k8s_runner.py", line 195, in run
10:42:11      res = self.do_query(milvus_instance, collection_name, top_ks, nqs, run_count, search_param)
10:42:11    File "/home/jenkins/agent/workspace/milvus-debug/milvus_benchmark/runner.py", line 200, in do_query
10:42:11      query_res = milvus.query(vectors, top_k, search_param=search_param)
10:42:11    File "/home/jenkins/agent/workspace/milvus-debug/milvus_benchmark/client.py", line 31, in wrapper
10:42:11      result = func(*args, **kwargs)
10:42:11    File "/home/jenkins/agent/workspace/milvus-debug/milvus_benchmark/client.py", line 145, in query
10:42:11      self.check_status(status)
10:42:11    File "/home/jenkins/agent/workspace/milvus-debug/milvus_benchmark/client.py", line 72, in check_status
10:42:11      raise Exception("Status not ok")
10:42:11  Exception: Status not ok
10:42:11  
```

```
Error, can't get gpu: -1resource
2020-03-18 00:16:55,806 | ERROR | default | [ENGINE] Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 8388608 bytes (error 2 out of memory)
2020-03-18 00:16:55,806 | ERROR | default | [WRAPPER] Error in virtual knowhere::DatasetPtr knowhere::IVF::Search(const DatasetPtr&, const Config&) at IndexIVF.cpp:139: Error in virtual void knowhere::IVFSQHybrid::search_impl(int64_t, const float*, int64_t, float*, int64_t*, const Config&) at IndexIVFSQHybrid.cpp:124: Hybrid Search Error, can't get gpu: -1resource
2020-03-18 00:16:55,806 | ERROR | default | [ENGINE] Search error:Error in virtual knowhere::DatasetPtr knowhere::IVF::Search(const DatasetPtr&, const Config&) at IndexIVF.cpp:139: Error in virtual void knowhere::IVFSQHybrid::search_impl(int64_t, const float*, int64_t, float*, int64_t*, const Config&) at IndexIVFSQHybrid.cpp:124: Hybrid Search Error, can't get gpu: -1resource
2020-03-18 00:16:55,846 | ERROR | default | [ENGINE] Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 8388608 bytes (error 2 out of memory)
2020-03-18 00:16:55,847 | ERROR | default | [WRAPPER] Error in virtual knowhere::DatasetPtr knowhere::IVF::Search(const DatasetPtr&, const Config&) at IndexIVF.cpp:139: Error in virtual void knowhere::IVFSQHybrid::search_impl(int64_t, const float*, int64_t, float*, int64_t*, const Config&) at IndexIVFSQHybrid.cpp:124: Hybrid Search Error, can't get gpu: -1resource
2020-03-18 00:16:55,848 | ERROR | default | [ENGINE] Search error:Error in virtual knowhere::DatasetPtr knowhere::IVF::Search(const DatasetPtr&, const Config&) at IndexIVF.cpp:139: Error in virtual void knowhere::IVFSQHybrid::search_impl(int64_t, const float*, int64_t, float*, int64_t*, const Config&) at IndexIVFSQHybrid.cpp:124: Hybrid Search Error, can't get gpu: -1resource
2020-03-18 00:16:55,852 | ERROR | default | [SERVER] Request failed with code: Error code(50003): Error in virtual knowhere::DatasetPtr knowhere::IVF::Search(const DatasetPtr&, const Config&) at IndexIVF.cpp:139: Error in virtual void knowhere::IVFSQHybrid::search_impl(int64_t, const float*, int64_t, float*, int64_t*, const Config&) at IndexIVFSQHybrid.cpp:124: Hybrid Search Error, can't get gpu: -1resource
```

**Steps/Code to reproduce behavior**
Run search with GPU-version, 
gpu-threshold: 200
gpu-cache-size: 4,6
nq: 200


**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details**
2 GPU-card(8G)

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

milvusdb/daily-build:ubuntu18.04-gpu-20200315   ==> can reproduce

Welcome to Milvus!
Milvus Release version: v0.7.0, built at 2020-03-15 12:47.26, with OpenBLAS library.
You are using Milvus GPU edition
Last commit id: 6c826c1308cfa0982652ccb687c87e6dd8c1b727

2020-03-20 17:41:51,668 | ERROR | default | [ENGINE] Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 8388608 bytes (error 2 out of memory)
2020-03-20 17:41:51,707 | ERROR | default | [ENGINE] Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 8388608 bytes (error 2 out of memory)
2020-03-20 17:41:51,712 | ERROR | default | [SERVER] Request failed with code: Error code(30117): out of memory: CPU2GPU:0


=====================================================================
milvusdb/daily-build:ubuntu18.04-gpu-20200314  ==> no this issue

Welcome to Milvus!
Milvus Release version: v0.7.0, built at 2020-03-14 10:22.02, with OpenBLAS library.
You are using Milvus GPU edition
Last commit id: 1b60e3bca6bd99c01f4825ba1d3c0275ca92ca1c

--- a/core/src/cache/CacheMgr.h
+++ b/core/src/cache/CacheMgr.h
@@ -57,12 +57,6 @@ class CacheMgr {
     void
     SetCapacity(int64_t capacity);
 
-    void
-    Lock();
-
-    void
-    Unlock();
-
  protected:
     CacheMgr();
 
@@ -70,7 +64,6 @@ class CacheMgr {
 
  protected:
     std::shared_ptr<Cache<ItemObj>> cache_;
-    std::mutex mutex_;
 };
 
 }  // namespace cache

--- a/core/src/cache/CacheMgr.inl
+++ b/core/src/cache/CacheMgr.inl
@@ -133,17 +133,5 @@ CacheMgr<ItemObj>::SetCapacity(int64_t capacity) {
     cache_->set_capacity(capacity);
 }
 
-template <typename ItemObj>
-void
-CacheMgr<ItemObj>::Lock() {
-    mutex_.lock();
-}
-
-template <typename ItemObj>
-void
-CacheMgr<ItemObj>::Unlock() {
-    mutex_.unlock();
-}
-
 }  // namespace cache
 }  // namespace milvus

--- a/core/src/cache/GpuCacheMgr.cpp
+++ b/core/src/cache/GpuCacheMgr.cpp
@@ -84,9 +84,7 @@ GpuCacheMgr::GetInstance(int64_t gpu_id) {
 void
 GpuCacheMgr::OnGpuCacheCapacityChanged(int64_t capacity) {
     for (auto& iter : instance_) {
-        iter.second->Lock();
         iter.second->SetCapacity(capacity * G_BYTE);
-        iter.second->Unlock();
     }
 }
 

--- a/core/src/db/engine/ExecutionEngineImpl.cpp
+++ b/core/src/db/engine/ExecutionEngineImpl.cpp
@@ -555,11 +555,9 @@ ExecutionEngineImpl::CopyToGpu(uint64_t device_id, bool hybrid) {
              */
             ENGINE_LOG_DEBUG << "CPU to GPU" << device_id << " start";
             auto gpu_cache_mgr = cache::GpuCacheMgr::GetInstance(device_id);
-            // gpu_cache_mgr->Lock();
             // gpu_cache_mgr->Reserve(index_->Size());
             index_ = knowhere::cloner::CopyCpuToGpu(index_, device_id, knowhere::Config());
             // gpu_cache_mgr->InsertItem(location_, std::static_pointer_cast<cache::DataObj>(index_));
-            // gpu_cache_mgr->Unlock();
             ENGINE_LOG_DEBUG << "CPU to GPU" << device_id << " finished";
         } catch (std::exception& e) {
             ENGINE_LOG_ERROR << e.what();
@@ -577,10 +575,8 @@ ExecutionEngineImpl::CopyToIndexFileToGpu(uint64_t device_id) {
     // the ToIndexData is only a placeholder, cpu-copy-to-gpu action is performed in
     if (index_) {
         auto gpu_cache_mgr = milvus::cache::GpuCacheMgr::GetInstance(device_id);
-        gpu_cache_mgr->Lock();
         gpu_num_ = device_id;
         gpu_cache_mgr->Reserve(index_->Size());
-        gpu_cache_mgr->Unlock();
     }
 #endif
     return Status::OK();
@@ -964,10 +960,8 @@ ExecutionEngineImpl::GetVectorByID(const int64_t& id, uint8_t* vector, bool hybr
 Status
 ExecutionEngineImpl::Cache() {
     auto cpu_cache_mgr = milvus::cache::CpuCacheMgr::GetInstance();
-    cpu_cache_mgr->Lock();
     cache::DataObjPtr obj = std::static_pointer_cast<cache::DataObj>(index_);
     cpu_cache_mgr->InsertItem(location_, obj);
-    cpu_cache_mgr->Unlock();
     return Status::OK();
 }
 

